---
title: "DREAM CPTAC: Protein weight analysis"
author: "MI YANG"
date: "`r doc_date()`"
vignette: >
  %\VignetteIndexEntry{Bioconductor style for HTML documents}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
output: 
  BiocStyle::html_document
---
```{r include=FALSE, cache=FALSE}
library(pheatmap)
path <- "~/Documents/RWTH_Aachen"
source(paste0(path,"/FUNCTIONS/general_functions.R" ))
source(paste0(path,"/FUNCTIONS/PLOT.R" ))
source(paste0(path,"/FUNCTIONS/survival_analysis_functions.R" ))
source(paste0(path,"/DREAM_CPTAC/PROT_PREDICTION/PROT_ANALYSIS_functions.R" ))
```


# IMPROVEMENT COMPARED TO PROTEIN/mRNA correlation

```{r include=FALSE, cache=FALSE}
path1 <- paste0(path,"/DREAM_CPTAC/CHALLENGE_RESULT/")
result_folder <- paste0(path,"/DREAM_CPTAC/PLOT/finalround/")

############ choose tissue ############
pred_ovarian <- read.csv(paste0(path1,"finalround/reguanteamspredictionfiles/guan_sc2_ovarian_cor.txt"), row.names=1, sep="")
protein_ovarian_mRNA_test <- read.delim(paste0(path,"/DREAM_CPTAC/DATA_prospective/SC2/prospective_ova_rna_seq_sort_common_gene_15121.txt"), row.names=1)
protein_ovarian_prot_test <- read.delim(paste0(path,"/DREAM_CPTAC/DATA_prospective/SC2/prospective_ova_proteome_sort_common_gene_7061.txt"), row.names=1)

L <- common_row(list(protein_ovarian_mRNA_test, protein_ovarian_prot_test))
corr_vect_ovarian <- corr_by_row(L[[1]],L[[2]])
corr_vect_ovarian_scored <- corr_vect_ovarian[ names(corr_vect_ovarian) %in% rownames(pred_ovarian) ]
mean(corr_vect_ovarian_scored)

pred_breast <- read.csv(paste0(path1,"finalround/reguanteamspredictionfiles/guan_sc2_breast_cor.txt"), row.names=1, sep="")
protein_breast_mRNA_test <- read.delim(paste0(path,"/DREAM_CPTAC/DATA_prospective/SC2/prospective_breast_RNA_sort_common_gene_15107.txt"), row.names=1)
protein_breast_prot_test <- read.delim(paste0(path,"/DREAM_CPTAC/DATA_prospective/SC2/prospective_breast_proteome_sort_common_gene_10005.txt"), row.names=1)

L <- common_row(list(protein_breast_mRNA_test, protein_breast_prot_test))
corr_vect_breast <- corr_by_row(L[[1]],L[[2]])
corr_vect_breast_scored <- corr_vect_breast[ names(corr_vect_breast) %in% rownames(pred_breast) ]
mean(corr_vect_breast_scored)

#######################################

```



# PROTEIN WEIGHT ANALYSIS (GUAN METHOD with significance assessment from Francesca)

# For BREAST

```{r include=FALSE, cache=FALSE}

Regulators_number_regression_breast <- read.csv(paste0(path,"/DREAM_CPTAC/PROT_PREDICTION/DATA_RESULT_STORAGE/COMMON_REGULATORS/Regulators_number_regression_breast.csv"))
Regulators <- Regulators_number_regression_breast$counts

# Kernel Density Plot
d <- density(Regulators) # returns the density data 
plot(d) # plots the results

##  1%, 5% and 10%
##  report survival for different cut-off
percent_1 <- quantile( Regulators, probs=seq( 0.90, 1 ) ) #  
top <- as.character( Regulators_number_regression_breast[ which(Regulators > as.numeric(percent_1) ) , 1] ) 
top_regulator <- top
## to make it clear ! 
total <- sum(Regulators) ; total_top_regulator <- sum(Regulators_number_regression_breast[ Regulators_number_regression_breast$regulator %in% top_regulator, 2 ] )


```


# test survival of top predictors (BREAST)

```{r include=FALSE, cache=FALSE}
path2 <- paste0(path,"/DREAM_CPTAC/CHALLENGE_RESULT/")
path3 <- paste0(path2,"00_PREDICTOR_WEIGHTS/")
result_folder <- paste0(path,"/DREAM_CPTAC/PLOT/finalround/REGULATORS")

tissue <- "breast"

# load data
version <- "05_version"
predicted_protein <- read.csv(paste0(path,"/DREAM_CPTAC/PROT_PREDICTION/DATA_POST_CHALLENGE/TCGA/",tissue,"/RNAseq_external/predicted_protein"), row.names=1, check.names = F )
mRNA <- read.delim(paste0(path,"/DREAM_CPTAC/PROT_PREDICTION/DATA_POST_CHALLENGE/TCGA/",tissue,"/RNAseq_external/prospective_breast_RNA_sort_common_gene_15107.txt"), row.names=1, check.names = F )
mRNA <- mRNA[ ,-1]

########################## functions ###########################
survival_analysis_continuous = function(CAT, CLINICS){ #  CLINICS=CLINICS_breast
  library(reshape)
  library(survminer)
  status = CLINICS$EVENT
  stime = CLINICS$DAYS_TO_LAST_FOLLOWUP
  var_01 = CLINICS$AGE
  var_02 = CLINICS$GENDER
  var_03 = CLINICS$`Basal.like`
  var_04 = CLINICS$LuminalB
  var_05 = CLINICS$`HER2.enriched`
  var_06 = CLINICS$`Normal.like`
  var_07 = CLINICS$T2
  var_08 = CLINICS$T3
  var_09 = CLINICS$T4
  var_10 = CLINICS$TX
  
  RESULTS = list()
  n.dead.cases = table(status)['1']
  n.dead.cases[ is.na(n.dead.cases) ] = 1

  if( n.dead.cases > 3 ) {
    for (predictor in colnames(CAT)){  ##  predictor="HSH2D"
      X = CAT[  , predictor  ]
      X <- as.numeric(X)  
      
      # mysurfit2 = coxph(Surv(stime, status) ~ X)  #  summary(mysurfit)
      if( length(unique(var_02)) > 1 ) # gender
        mysurfit2 = coxph(Surv(stime, status) ~ var_01+var_02+X)
      # var_01+var_02+var_03+var_04+var_05+var_06+var_07+var_08+var_09+var_10
      if( length(unique(var_02)) == 1 )# gender  
        mysurfit2 = coxph(Surv(stime, status) ~ age+X)   #  summary(mysurfit2)
      X_pval = summary(mysurfit2)$coefficients['X', 'Pr(>|z|)']  #  Cox p-value tests if strata is significantly associated with survival.
      X_coef = unname(coef(mysurfit2)['X'])
      RESULTS[[predictor]] = c(predictor= predictor, X_pval = X_pval, 'X_coef' = X_coef ) 
      # mysurvdiff = survdiff(Surv(stime, status) ~ X) # 
      X_pval[ is.na(X_pval) ] = 10
      #  summary( mysurfit2 ) ; basehaz(mysurfit2) predictSurvProb
      #  put here for 3 levels discretization
    }
    RESULTS = lapply(RESULTS, function(x) data.frame(t(data.frame(x, stringsAsFactors = F)), stringsAsFactors = F) )
    mRESULTS = melt(RESULTS, id.vars = colnames(RESULTS[[1]]) )
    names(mRESULTS)[ names(mRESULTS) == 'L1' ] = 'predictor'
    mRESULTS$X_adjpval = p.adjust(mRESULTS$X_pval, method = 'fdr')
  }
  mRESULTS <- mRESULTS[ ,c(1,2,5,3)]
#   fit <- coxph(Surv(stime, status) ~ age+gender+X)
#   plot(survfit(fit)) 
  return(mRESULTS)
}


survival_KM_plot <- function(data,title,size=25) {
  library(survminer)
  fit <- survfit( coxph ( Surv(DAYS_TO_LAST_FOLLOWUP, EVENT) ~ strata(Biomarker)+AGE+GENDER, data = data) )
  # fit <- survfit( Surv(DAYS_TO_LAST_FOLLOWUP, EVENT) ~ Biomarker, data = data )
  # http://www.sthda.com/english/wiki/survminer-r-package-survival-data-analysis-and-visualization
  size <- size
  ggsurvplot(
    fit,                  # survfit object with calculated statistics.
    size=2,               # line size
    censor.size=8,        # censor point size
    legend.labs = c("low", "high"),
    data = data,          # data used to fit survival curves.
    risk.table = F,       # show risk table.
    pval = F,             # show p-value of log-rank test.
    conf.int = F,         # show confidence intervals for 
    # point estimates of survival curves.
    # xlim = c(0,5000),   # present narrower X axis, but not affect
    # survival estimates.
    xlab = "Time in days",     # customize X axis label.
    break.time.by = 3000,      # break X axis in time intervals by 500.
    ggtheme = theme_light(),   # customize plot and risk table with a theme.
    risk.table.y.text.col = T, # colour risk table text annotations.
    risk.table.y.text = F,     # show bars instead of names in text annotations
    title = title,
    # in legend of risk table
    font.title=size, font.subtitle=size, font.caption=size, font.x=size, font.y=size, font.tickslab=size, font.legend=size 
  )  
}

############################################# check mRNA's predictability ############################################

CAT <- t(mRNA[top_regulator,  ])
CLINICS <- read.csv(paste0(path,"/TCGA/DATA/CLINICS_breast"), row.names=1)
common <- intersect(rownames(CLINICS),rownames(CAT))
CAT <- data.frame(CAT) ; CAT <- binarize_by_column(CAT, 0.5, binary = T)
CLINICS <- read.csv(paste0(path,"/TCGA/DATA/CLINICS_breast"), row.names=1)
CLINICS <- CLINICS[common, ] ; CAT <- CAT[common, ]

library(survival)
table <- survival_analysis_continuous (CAT=CAT, CLINICS=CLINICS ) ; table$X_pval <- as.numeric(table$X_pval) ; table
mRNA_of_survival <- which(table$X_adjpval < 0.25) ;  
mRNA_of_survival <- table$predictor[mRNA_of_survival]
table_subset <- table[table$predictor %in% mRNA_of_survival, ] ; length(which(table_subset$X_coef > 0))
df <- cbind(CAT, CLINICS)

for( i in mRNA_of_survival) {
  pdf(file=paste0(result_folder,"/version_significance/mRNA/KM_plot_mRNA_",paste0(i),".pdf"), width = 8 , height = 8, onefile = F )
  df_2 <- cbind(df[ ,i], df) ; colnames(df_2)[1] <- "Biomarker" ; 
  print( survival_KM_plot(data=df_2,  title=paste0(i,"  p=",formatC(table[table$predictor==i, 2], format = "e", digits = 1) ) , size = 32)  )  
  dev.off()
}

################################### check mRNA's predictability CONTINUOUS VERSION ##################################
CAT <- t(mRNA[top_regulator,rownames(CLINICS)])
CLINICS <- read.csv(paste0(path,"/TCGA/DATA/CLINICS_breast"), row.names=1)
CLINICS <- CLINICS[common, ] ; CAT <- CAT[common, ]

library(survival)
table <- survival_analysis_continuous (CAT=CAT, CLINICS=CLINICS ) ; table$X_pval <- as.numeric(table$X_pval) ; table
mRNA_of_survival <- which(table$X_adjpval < 0.25) ;  
mRNA_of_survival <- table$predictor[mRNA_of_survival]
table_subset <- table[table$predictor %in% mRNA_of_survival, ] ; length(rownames(table_subset))

## check overall all genes' coefficients.
CAT_mRNA <- t(mRNA) ; CAT_mRNA <- CAT_mRNA[ rownames(CAT) , colnames(CAT_mRNA) %not in% colnames(CAT)]
CLINICS <- read.csv(paste0(path,"/TCGA/DATA/CLINICS_breast"), row.names=1)
CLINICS <- CLINICS[common, ] ; CAT_mRNA <- CAT_mRNA[common, ]

table <- survival_analysis_continuous (CAT=CAT_mRNA, CLINICS=CLINICS ) ; table$X_pval <- as.numeric(table$X_pval) ; table
mRNA_of_survival <- which(table$X_adjpval < 0.25) ;  
mRNA_of_survival <- table$predictor[mRNA_of_survival]
table_subset <- table[table$predictor %in% mRNA_of_survival, ] ;  length(rownames(table_subset))

################################### top 10% regulators ###################################

top_hit_common_regulators <- 39
total_common_regulators   <- length(colnames(CAT))
top_hit_rest              <- 581
total_rest                <- length(colnames(CAT_mRNA))

mat <- cbind(c(top_hit_common_regulators,total_common_regulators-top_hit_common_regulators), c(top_hit_rest,total_rest - top_hit_rest - top_hit_common_regulators)) ; fisher.test(mat)  

```


# test ESSENTIALITY of top predictors (BREAST)

```{r include=FALSE, cache=FALSE}

result_folder <- paste0(path,"/DREAM_CPTAC/PLOT/finalround/REGULATORS")

Regulators_number_regression_breast <- read.csv(paste0(path,"/DREAM_CPTAC/PROT_PREDICTION/DATA_RESULT_STORAGE/COMMON_REGULATORS/Regulators_number_regression_breast.csv"))
Regulators_number_regression_ova <- read.csv(paste0(path,"/DREAM_CPTAC/PROT_PREDICTION/DATA_RESULT_STORAGE/COMMON_REGULATORS/Regulators_number_regression_ova.csv"))
Regulators_in_common <- intersect(Regulators_number_regression_breast$regulator,Regulators_number_regression_ova$regulator) ; length(Regulators_in_common)

# Essentiality score dowloaded from https://depmap.org/portal
RNAi_total <- read.csv(paste0(path,"/DATABASE/DEPMAP/RNAi/D2_combined_gene_dep_scores.csv"), row.names=1, check.names = F)
rownames(RNAi_total) <- sapply(strsplit(rownames(RNAi_total)," "), `[`, 1)
essentiality <- RNAi_total [ ,grep("BREAST", colnames(RNAi_total))] 
essentiality <- RNAi_total [ ,grep("OVARIAN", colnames(RNAi_total))]
essentiality_RNAi_total <- RNAi_total

CRISPR <- read.csv(paste0(path,"/DATABASE/DEPMAP/CRISPR/gene_effect.csv"), row.names=1, check.names = F); CRISPR <- t(CRISPR)
rownames(CRISPR) <- sapply(strsplit(rownames(CRISPR)," "), `[`, 1)
essentiality_CRISPR <- CRISPR

sample_info <- read.csv(paste0(path,"/DATABASE/DEPMAP/CRISPR/sample_info.csv"), check.names = F)

divide_into_tissue <- function(data, sample_info, column=2) {  # data=essentiality_regulator_ALL_CRISPR
  common <- intersect(sample_info[ ,column], colnames(data))
  data <- data[ ,common]
  sample_info <- sample_info[sample_info[ ,column] %in% common, ]
 
  to_exclude <- names(which(table(sample_info$primary_tissue) < 10))
  sample_info <- sample_info[ -which(sample_info$primary_tissue %in% to_exclude) , ]
  sample_info$primary_tissue <- as.character(sample_info$primary_tissue)
 
  mat_list <- list() 
  for(i in 1:length(table(sample_info$primary_tissue))) {  #  i=1
    tissue <- names(table(sample_info$primary_tissue))[i]
    data_tissue <- data[ ,colnames(data) %in% sample_info[sample_info$primary_tissue == tissue,column]]
    mat_list[[i]] <- as.numeric(data_tissue)
  }
  names(mat_list) <- names(table(sample_info$primary_tissue))
  return(mat_list)
}


## all regulators RNAi_total
essentiality_regulator_ALL_RNAi_total <- essentiality_RNAi_total[rownames(essentiality_RNAi_total) %in% Regulators_in_common, ]
essentiality_regulator_ALL_RNAi_total <- rowMeans(essentiality_regulator_ALL_RNAi_total, na.rm = T)

essentiality_not_regulator_RNAi_total <- essentiality_RNAi_total[rownames(essentiality_RNAi_total) %not in% Regulators_in_common, ]
essentiality_not_regulator_RNAi_total <- rowMeans(essentiality_not_regulator_RNAi_total, na.rm = T)

## all regulators CRISPR
essentiality_regulator_ALL_CRISPR <- essentiality_CRISPR[rownames(essentiality_CRISPR) %in% Regulators_in_common, ]
essentiality_regulator_ALL_CRISPR <- divide_into_tissue(essentiality_regulator_ALL_CRISPR, sample_info)

essentiality_not_regulator_CRISPR <- essentiality_CRISPR[rownames(essentiality_CRISPR) %not in% Regulators_in_common, ]
essentiality_not_regulator_CRISPR <- divide_into_tissue(essentiality_not_regulator_CRISPR, sample_info)

#   t.test(essentiality_regulator_ALL_RNAi_total, essentiality_not_regulator_RNAi_total)
#   t.test(essentiality_regulator_ALL_CRISPR, essentiality_not_regulator_CRISPR)
# result <- list(essentiality_regulator_ALL_RNAi_total, essentiality_not_regulator_RNAi_total,essentiality_regulator_ALL_CRISPR, essentiality_not_regulator_CRISPR)
# names(result) <- c("Common predictors (RNAi)","Other genes (RNAi)","Common predictors (CRISPR)","Other genes (CRISPR)" )

result_1 <- essentiality_regulator_ALL_CRISPR
result_2 <- essentiality_not_regulator_CRISPR

nice_boxplot_list <- function (result_1, result_2, title="www", text_size=25, title_size=2, Y_label="Dependency Score (Essentiality)" ,X_label="Tissues") {
  
  result <- result_1
  result_transformed <- c()
  for (i in 1:length(result)) {
    v <- na.omit(result[[i]]) ; v <- cbind(rep(i,each=length(na.omit(result[[i]]))), v)
    result_transformed <- rbind(result_transformed, v)
    colnames(result_transformed) <- c("Tissue", "Essentiality")
  }
  result_transformed_1 <- data.frame(result_transformed)
  
  result <- result_2
  result_transformed <- c()
  for (i in 1:length(result)) {
    v <- na.omit(result[[i]]) ; v <- cbind(rep(i,each=length(na.omit(result[[i]]))), v)
    result_transformed <- rbind(result_transformed, v)
    colnames(result_transformed) <- c("Tissue", "Essentiality")
  }
  result_transformed_2 <- data.frame(result_transformed)
  
  result_transformed <- result_transformed_1
  result <- result_1
  result_transformed <- cbind(rep(NA,length(result_transformed[ ,1])), result_transformed)
  colnames(result_transformed)[1] <- "subset"
  result_transformed[ , 1]  <- 1
  result_transformed$Tissue <- factor(result_transformed$Tissue,levels=1:length(names(result)), labels=names(result)) 
  result_transformed_1 <- result_transformed
  
  result_transformed <- result_transformed_2
  result <- result_2
  result_transformed <- cbind(rep(NA,length(result_transformed[ ,1])), result_transformed)
  colnames(result_transformed)[1] <- "subset"
  result_transformed[ , 1]  <- 2
  result_transformed$Tissue <- factor(result_transformed$Tissue,levels=1:length(names(result)), labels=names(result)) 
  result_transformed_2 <- result_transformed
  
  result_transformed <- rbind(result_transformed_1,result_transformed_2)
  result_transformed$subset <- factor(result_transformed$subset,levels=1:2, labels=c("Common predictors", "Other genes") ) 
  r <- ggplot(data = result_transformed, aes(y = Essentiality, x = Tissue, fill = subset )) + geom_boxplot(width = 0.7 , colour="black",outlier.size = 0.05) + 
    ggtitle(title) + ylab(Y_label) + xlab(X_label) + theme(legend.position="right",legend.title=element_text(size=text_size-5),legend.text=element_text(size=text_size-5),legend.key.size = unit(2,"line"), plot.title=element_text(size=text_size,hjust=0.5),axis.text=element_text(colour="black",size=text_size ), axis.title.x=element_text(colour="black",size=text_size),axis.title.y=element_text(colour="black",size=text_size),panel.background = element_rect(fill='white'),panel.grid.major = element_line(colour = "grey90") )  + scale_y_continuous( breaks=c(-2,-1, 0,1,1.5), limits=c(-2, 1.2))  + coord_flip()
  grid.arrange(r, ncol = 1)  
}

tiff(file=paste0(result_folder,"/essentiality_all_tissues.tiff"), width = 1000, height = 1000 )
nice_boxplot_list(result_1, result_2, title="Broad DepMap gene essentialities")
dev.off()


```



[R]: http://r-project.org
[RStudio]: http://www.rstudio.com/